### PR TITLE
added Java ArrayList: .trimToSize()

### PR DESCRIPTION
**Pull request by ATREAY**
### Description

This pull request introduces a new term, `trimToSize()`, within the Java ArrayList section.
fixes: #4106

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/array-list/terms/trimToSize/trimToSize.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+Title: '.trimToSize()'
+Description: 'Adjusts the capacity of the ArrayList to be the same as its size.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Data Types'
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/trimToSize/trimToSize.md:9]:
```suggestion
  - 'Data Types'
  - 'Methods'
```

File: content/java/concepts/array-list/terms/trimToSize/trimToSize.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+The `.trimToSize()` method does not require any parameters. It adjusts the capacity of the `ArrayList` to be the same as its size:
+
+```pseudo
+list.trimToSize();
+```
+
+## Example
+
+The following example creates an `ArrayList`, uses [`.add()`](https://www.codecademy.com/resources/docs/java/array-list/add) to append elements to it, and then trims its capacity to match its size using `.trimToSize()`:
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/trimToSize/trimToSize.md:27]:
```suggestion
The following example creates an `ArrayList`, uses [`.add()`](https://www.codecademy.com/resources/docs/java/array-list/add) method to append elements to it, and then trims its capacity to match its size using `.trimToSize()`:
```

File: content/java/concepts/array-list/terms/trimToSize/trimToSize.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    list.add("Banana");
+    list.add("Orange");
+
+    // Trim the capacity to match the size
+    list.trimToSize();
+
+    // Display the trimmed capacity and size
+    System.out.println("Trimmed Capacity: " + list.size());
+  }
+}
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/trimToSize/trimToSize.md:48]:
```suggestion
import java.util.ArrayList;

public class Main {
  public static void main(String[] args) {
    // Create an ArrayList of Strings with an initial capacity
    ArrayList<String> list = new ArrayList<>(10);

    // Add some elements to the list
    list.add("Apple");
    list.add("Banana");
    list.add("Orange");

    // Trim the capacity to match the size
    list.trimToSize();
    
    System.out.println("Trimmed Capacity: " + list.size());
  }
}
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few changes from my side for you @ATREAY 🚀

**Comment by ATREAY** [comment]
Hey @dakshdeepHERE, I have made the required changes, please go through it when you have a moment.

**Review by dakshdeepHERE — APPROVED** [review]
Looks good for second review 🚀

File: content/java/concepts/array-list/terms/trimToSize/trimToSize.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Data Types'
+  - 'Methods'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`.trimToSize()`** method of the `ArrayList` class adjusts the capacity of the list to be the same as its size.
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/trimToSize/trimToSize.md:15]: Hyperlink the `ArrayList` with [this](https://www.codecademy.com/resources/docs/java/array-list) docs url.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@ATREAY I have suggested a small change, please make it.

**Comment by ATREAY** [comment]
Hey @avdhoottt, I have made the suggested changes, please go through it when you have a moment.

**Review by avdhoottt — APPROVED** [review]
Looks good to get merged.

**Pull request merged**